src/Ocelot/Cache/CacheKeyGenerator.cs

```diff
@@ -1,20 +1,43 @@
-﻿using Ocelot.Request.Middleware;
+﻿using Ocelot.Configuration;
+using Ocelot.Request.Middleware;
 
 namespace Ocelot.Cache
 {
     public class CacheKeyGenerator : ICacheKeyGenerator
     {
-        public string GenerateRequestCacheKey(DownstreamRequest downstreamRequest)
+        private const char Delimiter = '-';
+        
+        public async ValueTask<string> GenerateRequestCacheKey(DownstreamRequest downstreamRequest, DownstreamRoute downstreamRoute)
         {
-            var downStreamUrlKeyBuilder = new StringBuilder($"{downstreamRequest.Method}-{downstreamRequest.OriginalString}");
-            if (downstreamRequest.Content != null)
+            var builder = new StringBuilder()
+                .Append(downstreamRequest.Method)
+                .Append(Delimiter)
+                .Append(downstreamRequest.OriginalString);
+
+            var cacheOptionsHeader = downstreamRoute?.CacheOptions?.Header;
+            if (!string.IsNullOrEmpty(cacheOptionsHeader))
+            {
+                var header = downstreamRequest.Headers
```

Comment:
I'm not sure it's really what we want. Here we are using the header value to create the MD5 key. I'm not sure the design is safe. Besides, no acceptance tests were provided by the author, so I'm concerned about that change.

Reply:
Design **wasn't** even not discussed by the team. Raynald had provided review and gave approval.
The author just wanted to merge it. I was not careful allowing to merge without acceptance tests...
We have an incident with dev process violation. This is my fault!